tests/regression/anomaly/test_anomaly_classificaiton.py

```diff
@@ -0,0 +1,240 @@
+)
+
+# Configurations for regression test.
+TASK_TYPE = "anomaly_classification"
+SAMPLED_ANOMALY_DATASET_CATEGORIES = random.sample(ANOMALY_DATASET_CATEGORIES, 1)
+
+otx_dir = os.getcwd()
+templates = Registry("otx/algorithms/anomaly").filter(task_type=TASK_TYPE.upper()).templates
+templates_ids = [template.model_template_id for template in templates]
+
+result_dict = get_result_dict(TASK_TYPE)
+result_dir = f"/tmp/regression_test_results/{TASK_TYPE}"
+Path(result_dir).mkdir(parents=True, exist_ok=True)
+
+# Detection
+anomaly_classification_regression_config = load_regression_configuration(otx_dir, TASK_TYPE)
+anomaly_classification_data_args = anomaly_classification_regression_config["data_path"]
+
+
+class TestRegressionAnomalyClassification:
+    def setup_method(self):
+        self.performance = {}
+
+    def teardown_method(self):
+        with open(f"{result_dir}/result.json", "w") as result_file:
+            json.dump(result_dict, result_file, indent=4)
+
+    def _apply_category(self, data_dict, category):
+        return_dict = {}
+        for k, v in data_dict.items():
+            if "train" in k:
+                return_dict[k] = f"{v}/{category}/train"
+            if "val" in k or "test" in k:
+                return_dict[k] = f"{v}/{category}/test"
+            else:
+                return_dict[k] = v
+        return return_dict
+
+    @e2e_pytest_component
+    @pytest.mark.parametrize("template", templates, ids=templates_ids)
+    @pytest.mark.parametrize("category", SAMPLED_ANOMALY_DATASET_CATEGORIES)
+    # ANOMALY_DATASET_CATEGORIES : all categories of MVTec dataset.
+    def test_otx_train(self, template, tmp_dir_path, category):
+        self.performance[template.name] = {}
+        category_data_args = self._apply_category(anomaly_classification_data_args, category)
+
+        tmp_dir_path = tmp_dir_path / TASK_TYPE
+        train_start_time = timer()
+        otx_train_testing(template, tmp_dir_path, otx_dir, category_data_args)
+        train_elapsed_time = timer() - train_start_time
+
+        infer_start_time = timer()
+        otx_eval_compare(
+            template,
+            tmp_dir_path,
+            otx_dir,
+            category_data_args,
+            anomaly_classification_regression_config["regression_criteria"]["train"][category],
+            self.performance[template.name],
+        )
+        infer_elapsed_time = timer() - infer_start_time
+
+        self.performance[template.name][TIME_LOG["train_time"]] = round(train_elapsed_time, 3)
+        self.performance[template.name][TIME_LOG["infer_time"]] = round(infer_elapsed_time, 3)
+        result_dict[TASK_TYPE]["train"][category].append(self.performance)
+
+    @e2e_pytest_component
+    @pytest.mark.parametrize("template", templates, ids=templates_ids)
+    @pytest.mark.parametrize("category", SAMPLED_ANOMALY_DATASET_CATEGORIES)
+    def test_otx_train_kpi_test(self, template, category):
+        results = result_dict[TASK_TYPE]["train"][category]
```

Comment:
Can you add a docstring explaining what this test does?

Reply:
Done